src/main/java/de/retest/recheck/cli/subcommands/Diff.java

```diff
@@ -1,73 +1,169 @@
+			return false;
+		}
+		return true;
+	}
+
+	private ActionReplayResult createActionReplayResultFrom( final Path expectedSutStatePath,
+			final Path actualSutStatePath ) {
+		final SutState expectedSutState = goldenMasterProvider
+				.loadGoldenMaster( createFile( expectedSutStatePath, RecheckProperties.DEFAULT_XML_FILE_NAME ) );
+		final SutState actualSutState = goldenMasterProvider
+				.loadGoldenMaster( createFile( actualSutStatePath, RecheckProperties.DEFAULT_XML_FILE_NAME ) );
+
+		return getActionReplayResult( expectedSutStatePath.toString(), expectedSutState, actualSutState );
+	}
+
+	private ActionReplayResult createFilteredActionReplayResult( final ActionReplayResult actionReplayResult ) {
 		final Filter excludeFilter = FilterUtil.getExcludeFilterFiles( exclude );
-		final Filter recheckIgnore = FilterUtil.loadRecheckIgnore();
-		final TestReportFilter filter = new TestReportFilter( new CompoundFilter( excludeFilter, recheckIgnore ) );
-		final TestReport filteredTestReport = filter.filter( report );
-		final TestReportPrinter printer = new TestReportPrinter( none() );
+		final TestReportFilter reportFilter =
+				new TestReportFilter( (excludeFilter) );
+		return reportFilter.filter( actionReplayResult );
+	}
+
+	private void persistIfOutputOptionPresent( final ActionReplayResult actionReplayResult ) throws IOException {
+		if ( directory != null ) {
+			persist( actionReplayResult, directory );
+			return;
+		}
+		logger.info( "\nThe displayed differences may be saved as test report by using the --output option." );
+	}
+
+	private File createFile( final Path path, final String defaultXmlFileName ) {
+		return new File( path.toString(), defaultXmlFileName );
+	}
+
+	private void printGoldenMasterDifferences( final ActionReplayResult actionReplayResult ) {
+		final ActionReplayResultPrinter printer =
+				new ActionReplayResultPrinter( ( attributes, s, serializable ) -> false );
 
-		logger.info( "\n{}", printer.toString( filteredTestReport ) );
+		logger.info( "\n{}", printer.toString( actionReplayResult ) );
 
-		if ( filteredTestReport.getDifferencesCount() > 0 ) {
+		final int differenceCount = actionReplayResult.getDifferences().size();
+		if ( differenceCount > 0 ) {
 			logger.info( "\nOverall, recheck found {} difference(s) when checking {} element(s).",
-					filteredTestReport.getDifferencesCount(), filteredTestReport.getCheckedUiElementsCount() );
+					differenceCount, actionReplayResult.getCheckedUiElementsCount() );
 		}
 	}
 
+	private ActionReplayResult getActionReplayResult( final String expectedSutStatePath,
+			final SutState expectedSutState,
+			final SutState actualSutState ) {
+		final RecheckDifferenceFinder differenceFinder =
+				new RecheckDifferenceFinder( ( attributes, s, serializable ) -> false,
+						"Comparison of Golden Masters",
+						expectedSutStatePath );
+		return differenceFinder.findDifferences( expectedSutState, actualSutState );
+	}
+
+	@SuppressWarnings( "ResultOfMethodCallIgnored" )
```

Comment:
Please ignore this within IntelliJ as this is quite useless in my opinion as in most cases that is more the fault of bad API design.